src/stackmap_checker/call_stack_state.c

```diff
@@ -219,22 +220,18 @@ void restore_unopt_stack(stack_map_t *sm, call_stack_state_t *state)
     free_locations(locations, num_locations);
 }
 
-void append_record(call_stack_state_t *state, stack_map_record_t first_rec)
-{
-    // The first stack map record to be stored is the one associated with the
-    // patchpoint which triggered the guard failure (so it needs to be added
-    // separately).
-    state->records = realloc(state->records,
-                             state->depth * sizeof(stack_map_record_t));
-    memmove(state->records + 1, state->records,
-            (state->depth - 1) * sizeof(stack_map_record_t));
-    state->records[0] = first_rec;
-}
-
 void restore_register_state(call_stack_state_t *state, uint64_t r[])
 {
     for (size_t i = 0; i < 16; ++i) {
-        r[i] = (uint64_t)state->registers[0][i];
+        r[i] = (uint64_t)state->frames[0].registers[i];
     }
 }
 
+frame_t* alloc_empty_frames(size_t num_frames)
+{
+    frame_t *frames = calloc(num_frames, sizeof(frame_t));
+    for (size_t i = 0; i < num_frames; ++i) {
+        frames[i].registers = calloc(16, sizeof(unw_word_t));
```

Comment:
why 16?

Reply:
This only restores the 16 general-purpose registers of `x86-64`.